python/pyarrow/serialization.py

```diff
@@ -76,6 +77,35 @@ def _deserialize_numpy_matrix(data):
         return np.matrix(data[0], dtype=np.dtype(data[1]), copy=False)
 
 
+# ----------------------------------------------------------------------
+# Set up serialization for scipy sparse matrices. Primitive types are handled
+# efficiently with Arrow's SparseTensor facilities, see numpy_convert.cc)
+
+def _serialize_scipy_sparse(obj):
+    if isspmatrix_coo(obj):
+        coords = np.vstack([obj.row, obj.col]).T
+        return pyarrow.SparseTensorCOO.from_numpy(obj.data, coords, obj.shape)
```

Comment:
do you need to return a tuple here with the sparse type? (at least that seems the way you expect the data in deserialize)

Reply:
Thanks for looking at this Joris!
You mean why don't we do: `pyarrow.SparseTensorCOO.from_numpy(obj.data, obj.row, obj.col, obj.shape)`? See: https://github.com/apache/arrow/pull/4446#pullrequestreview-256390267

Reply:
I meant the actual return value, not how the SparseTensor is created. Eg in the deserialize, you do ``data[0] == 'coo'``

Reply:
Oh, yeah, true! Thanks!
The plan was I write the basic python test first and then do the serialization next and then get back to python again.